orderbook/src/orderbook.rs

```diff
@@ -85,6 +89,52 @@ impl OrderBook {
         orders.retain(|_, order| has_future_valid_to(now_in_epoch_seconds, &order.order_creation));
     }
 
+    async fn remove_settled_orders(&self, settlement_contract: &GPv2Settlement) {
+        let uids_to_be_removed = self.get_uids_of_settled_orders(settlement_contract).await;
+        info!(
+            "removing following settled orders from orderbook: {:?}",
+            uids_to_be_removed
+        );
+        let mut orders = self.orders.write().await;
+        orders.retain(|uid, _| uids_to_be_removed.contains(uid));
+    }
+
+    async fn get_uids_of_settled_orders(
+        &self,
+        settlement_contract: &GPv2Settlement,
+    ) -> Vec<OrderUid> {
+        let orders = self.orders.read().await;
+        let uid_futures = orders.iter().map(|(uid, _)| async move {
+            (
+                *uid,
+                self.has_not_yet_been_settled(*uid, settlement_contract)
+                    .await,
+            )
+        });
+        let uid_pairs: Vec<(OrderUid, bool)> = join_all(uid_futures).await;
+        uid_pairs
+            .iter()
+            .filter(|(_, not_yet_settled)| *not_yet_settled)
+            .map(|(uid, _)| *uid)
+            .collect()
+    }
+
+    async fn has_not_yet_been_settled(
+        &self,
+        uid: OrderUid,
+        settlement_contract: &GPv2Settlement,
+    ) -> bool {
+        let filled_amount = settlement_contract
+            .filled_amount(uid.0.to_vec())
+            .call()
+            .await
+            .unwrap_or(U256::MAX);
+        // As a simplification the function is returning false,
```

Comment:
This requires adding a filled amount to the order user data. Could you create a task for it (probably a nice basic addition to add soon)?

Reply:
https://github.com/gnosis/gp-v2-services/issues/155